### ISSUE TITLE
"Out of range" error during database migration

### ISSUE THREAD

**Issue by 9662** [nextcloud/server#8143]
### Steps to reproduce
1. Find a 30Gb Nextcloud installation with a few thousand files and using SQLite as the database engine.
2. Decide that you are going to migrate that to MySQL.
3. Run something like `./occ db:convert-type --clear-schema --all-apps  mysql user host dbname`

### Expected behaviour
Conversion should proceed and complete successfully

### Actual behaviour
```
                                                                                                                                                                                                                                            
  [Doctrine\DBAL\Exception\DriverException]                                                                                                                                                                                                 
  An exception occurred while executing 'INSERT INTO `oc_filecache` (`fileid`, `storage`, `path_hash`, `parent`, `name`, `mimetype`, `mimepart`, `mtime`, `storage_mtime`, `encrypted`, `etag`, `path`, `permissions`, `size`, `unencrypte  
  d_size`, `checksum`) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params ["1", "1", "d41d8cd98f00b204e9800998ecf8427e", "-1", "", "2", "1", "4294967295", "1500726034", "0", "5a732839aae0a", "", "23", "8165566285", "0  
  ", null]:                                                                                                                                                                                                                                 
  SQLSTATE[22003]: Numeric value out of range: 1264 Out of range value for column 'mtime' at row 1                                                                                                                                          
                                                                                                                                                                                                                                            

                                                                                                    
  [Doctrine\DBAL\Driver\PDOException]                                                               
  SQLSTATE[22003]: Numeric value out of range: 1264 Out of range value for column 'mtime' at row 1  
                                                                                                    

                                                                                                    
  [PDOException]                                                                                    
  SQLSTATE[22003]: Numeric value out of range: 1264 Out of range value for column 'mtime' at row 1  
```

Indeed, on a mysql client:
```
INSERT INTO `oc_filecache` (`fileid`, `storage`, `path_hash`, `parent`, `name`, `mimetype`, `mimepart`, `mtime`, `storage_mtime`, `encrypted`, `etag`, `path`, `permissions`, `size`, `unencrypted_size`, `checksum`) VALUES("1", "1", "d41d8cd98f00b204e9800998ecf8427e", "-1", "", "2", "1", "4294967295", "1500726034", "0", "5a732839aae0a", "", "23", "8165566285", "0", null);
```

results in:

```
ERROR 1264 (22003): Out of range value for column 'mtime' at row 1
```


### Server configuration

**Operating system**: openSUSE Leap 42.2

**Web server:** Apache

**Database:** SQLite migrating to MySQL

**PHP version:** 7.2.0

**Nextcloud version:** 13.0.0. RC4

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from OC 6 or so

**Where did you install Nextcloud from:** Via ZIP package

### See also

#7637

**Comment by MorrisJobke** [comment]
Fixed by #8412

**Comment by tanzhenchao** [comment]
We found that adding the "unsigned" parameter can execute correctly, but we don't know where to modify the upgrade script.
`
ALTER TABLE oc_filecache CHANGE `path` `path` VARCHAR(4000) DEFAULT NULL, CHANGE `path_hash` `path_hash` VARCHAR(32) DEFAULT '' NOT NULL, CHANGE `name` `name` VARCHAR(250) DEFAULT NULL, CHANGE `mtime` `mtime` INT unsigned DEFAULT 0 NOT NULL, CHANGE `storage_mtime` `storage_mtime` INT unsigned DEFAULT 0 NOT NULL, CHANGE `etag` `etag` VARCHAR(40) DEFAULT NULL, CHANGE `checksum` `checksum` VARCHAR(255) DEFAULT NULL`
